File: solutions/day01/part2.py
from pathlib import Path

import numpy as np

module_path = Path(__file__).resolve().parent


def part_2(arr):
    solution = 0
    for num in arr[:, 0]:
        solution += num * (arr[:, 1] == num).sum()
    return solution


if __name__ == "__main__":
    input_arr = np.loadtxt(module_path / "input", dtype=int)
    solution = part_2(input_arr)
    print("Advent of Code 2024 (day 1)")
    print(f"Solution for part 2: {solution}")

File: solutions/day01/part2.c
#include <stdio.h>
#include <stdlib.h>


int main() {
    FILE *input;
    input = fopen("input", "r");

    unsigned int num, count;
    unsigned int sol2 = 0;

    // wc -l input
    unsigned int input_arr_1[1000], input_arr_2[1000];


    for (size_t i=0; i<1000; ++i) {
        fscanf(input, "%u   %u", &input_arr_1[i], &input_arr_2[i]);
    }

    for (size_t i=0; i<1000; ++i) {
        count = 0;
        num = input_arr_1[i];
        for (size_t j=0; j<1000; ++j) {
            if (num == input_arr_2[j]) count++;
        }
        sol2 += num * count;
    }

    printf("Advent of Code 2024 (day 1)\n");
    printf("Solution for part 2: %u\n", sol2);
}
